src/main/java/net/earthcomputer/clientcommands/features/MappingsHelper.java

```diff
@@ -0,0 +1,226 @@
+package net.earthcomputer.clientcommands.features;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import net.fabricmc.loader.api.FabricLoader;
+import net.fabricmc.mappingio.MappingReader;
+import net.fabricmc.mappingio.format.MappingFormat;
+import net.fabricmc.mappingio.tree.MappingTree;
+import net.fabricmc.mappingio.tree.MemoryMappingTree;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.StringReader;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Optional;
+
+public class MappingsHelper {
+
+    private static final MemoryMappingTree mojmapOfficial = new MemoryMappingTree();
+    private static final int SRC_OFFICIAL = 0;
+    private static final int DEST_OFFICIAL = 0;
+
+    private static final MemoryMappingTree officialIntermediaryNamed = new MemoryMappingTree();
+    private static final int SRC_INTERMEDIARY = 0;
+    private static final int DEST_INTERMEDIARY = 0;
+    private static final int SRC_NAMED = 1;
+    private static final int DEST_NAMED = 1;
+
+    private static final boolean IS_DEV_ENV = FabricLoader.getInstance().isDevelopmentEnvironment();
+
+    private static final HttpClient httpClient = HttpClient.newHttpClient();
```

Comment:
This could be a local variable, just so the resources for the http client are cleaned up after they are no longer needed.